steps/qaoa.py

```diff
@@ -57,8 +61,17 @@ def get_graph_partition_hamiltonian(graph):
     hamiltonian = partition.get_graph_partition_hamiltonian(graph_object)
     save_qubit_operator(hamiltonian, "hamiltonian.json")
 
-    
+
 def convert_relaxed_solution_to_angles(solution, epsilon=0.5):
     solution = np.array(load_list(solution))
     thetas = warm_start_ansatz.convert_relaxed_solution_to_angles(solution, epsilon)
     save_list(thetas.tolist(), "thetas.json")
+
+
+def get_problem_hamiltonian_from_graph(graph, problem_type):
+    if problem_type == "partition":
+        get_graph_partition_hamiltonian(graph)
+    elif problem_type == "maxcut":
+        get_maxcut_hamiltonian(graph)
+    else:
+        raise ValueError(f"Invalid problem type: {problem_type}.")
```

Comment:
I think this would be better represented as two functions, with typed input/return values:
1. `get_graph_partition_hamiltonian`
2. `get_maxcut_hamiltonian`

Rationale:
- Blocks of conditional logic is never ideal and usually leads to more maintenance cost
- Isolating the functions creates clear separation and easier to unit test (when the day comes)
- If we ever need to add any args to either function, we don't end up tainting the other
- This removes the `problem_type` argument and the function name becomes the key describing which problem type you have

Reply:
For the "production code in our library", I would agree.
But this is step that would be part of a workflow, so I intentionally want to keep it flexible and make it easier for someone to use in a workflow. Cause with this one can just iterate over problem types and doesn't need to write conditional logic for calling various python functions.

Reply:
Let's assume all these functions have the same interface – If they wouldn't we'd have problem in the dispatch code anyway. An Orquestra user wants to swap different Hamiltonian generators. With @mstechly's approach this requires changing `"partition"` workflow constant to `"maxcut"`. But is it much improvement over @simonwa7's approach where the user would simply change `"get_graph_partition_hamiltonian"` to `"get_maxcut_hamiltonian"` in the step function name section?

Reply:
Not sure if that's a valid argument, but for me iterating over arguments to a step in a workflow just feels better than iterating over the python function called.
Meeeh, I really don't know, I'm fine either way ;)